src/cupbearer/detectors/anomaly_detector.py

```diff
@@ -89,7 +89,8 @@ def _finetune(self, **kwargs) -> dict:
 
     def eval(
         self,
-        dataset: MixedData,
+        dataset: MixedData | None, 
```

Comment:
Is not having a default value for this deliberate? Seems a bit cumbersome if used with a dataloader. If we make both optional, then we should probably have a meaningful error message for when neither is provided.

Reply:
Nah I agree, adding a default argument seems better